pennylane/transforms/classical_jacobian.py

```diff
@@ -137,7 +139,12 @@ def classical_preprocessing(*args, **kwargs):
         """Returns the trainable gate parameters for a given QNode input."""
         trainable_only = kwargs.pop("_trainable_only", True)
         qnode.construct(args, kwargs)
-        return qml.math.stack(qnode.qtape.get_parameters(trainable_only=trainable_only))
+        tape = qnode.qtape
+
+        if expand_fn is not None:

```

Comment:
What if a user doesn't pass an expansion function, but there is a template? Should we check for this case and perform an expansion, or raise a warning?

Reply:
this is automatic, because the functions that call this one have default expansion functions :)